spec/unit/configurer_spec.rb

```diff
@@ -687,15 +677,18 @@ def expects_neither_new_or_cached_catalog
       end
 
       it "should return nil when the catalog's environment doesn't match the agent specified environment" do
-        configurer.instance_variable_set(:@environment, 'second_env')
+        Puppet[:environment] = 'second_env'
+        configurer = Puppet::Configurer.new
```

Comment:
Yep, unfortunately the `Configurer.new` doesn't provide a way to pass the environment in, and creating a new instance is the only way to be sure it's in the correct environment. For example, if I referenced `configurer` in a `before :each` block, then it would cause it to be created in the wrong environment (before the test had a chance to set `Puppet[:environment] = `